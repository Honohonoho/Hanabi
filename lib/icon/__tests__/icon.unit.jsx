import * as renderer from "react-test-renderer";
import {mount} from 'enzyme';
import React from "react";
import Icon from "../icon";

describe("icon", () => {
    it("icon render successfully", function () {
        const json = renderer.create(<Icon name="wechat"/>).toJSON();
        expect(json).toMatchSnapshot()
    });
    it('icon on click', () => {
        const fn = jest.fn();
        const component = mount(<Icon name="wechat" onClick={fn}/>);
        component.find('svg').simulate('click');
        expect(fn).toBeCalled()
    });
});
