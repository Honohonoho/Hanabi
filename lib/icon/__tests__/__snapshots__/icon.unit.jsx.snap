// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon icon render successfully 1`] = `
<svg
  className="h-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
